engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/types/TypePropagation.java

```diff
@@ -439,7 +443,12 @@ private TypeRepresentation processUnresolvedSymbolApplication(
             encounteredNoSuchMethod(
                 relatedWholeApplicationIR, argumentType, function.name(), MethodCallKind.STATIC);
           }
-          return resolvedStaticMethod;
+          if (isStaticMethodInvocation) {
+            var withSelf = TypeRepresentation.buildStaticMethod(typeObject, resolvedStaticMethod);
```

Comment:
- there is failing [Test org.enso.compiler.test.TypeInferenceTest.staticCallWithWrongType](https://github.com/enso-org/enso/actions/runs/18991427221/job/54245455267#step:14:6273) test
- I think it should be fixed by 6b5f1917c972c410f449614678bca8f1402890b5
- which prepends the "self type" as an argument in case of _static invocation_